### PR TITLE
Fix Calendar accessibility issues

### PR DESCRIPTION
**Pull request by DokaRus**
https://trello.com/c/9F1liQ9s

### CODE REVIEW

**Review by alexander-kotov-dx — DISMISSED** [review]

File: js/ui/calendar/ui.calendar.base_view.js [hunk 1]
```diff
@@ -100,70 +99,85 @@ const BaseView = Widget.inherit({
         let cellDate = this._getFirstCellData();
         let row;
 
-        function renderCell(cellIndex) {
+        const renderCell = (cellIndex) => {
```
Comment (San4es) [inline js/ui/calendar/ui.calendar.base_view.js]: Move to proto like `_createCell`?

Comment (DokaRus) [reply js/ui/calendar/ui.calendar.base_view.js]: Done

File: js/ui/calendar/ui.calendar.base_view.js [hunk 2]
```diff
@@ -86,49 +86,39 @@ const BaseView = Widget.inherit({
...
-        let row;
-
-        const renderCell = (cellIndex) => {
-            // T425127
-            if(prevCellDate) {
-                fixTimezoneGap(prevCellDate, cellDate);
-            }
-
-            prevCellDate = cellDate;
+        for(let indexRow = 0, len = this.option('rowCount'); indexRow < len; indexRow++) {
```
Comment (San4es) [inline js/ui/calendar/ui.calendar.base_view.js]:
```suggestion
        for(const rowIndex = 0, rowCount = this.option('rowCount'); rowIndex < rowCount; rowIndex++) {
```

File: js/ui/calendar/ui.calendar.base_view.js [hunk 3]
```diff
@@ -86,49 +86,39 @@ const BaseView = Widget.inherit({
...
-                cellTemplate.render(this._prepareCellTemplateData(cellDate, cellIndex, $cell));
-            } else {
-                cell.innerHTML = this._getCellText(cellDate);
-            }
+        return row;
+    },
 
-            cellDate = this._getNextCellData(cellDate);
-        };
+    _appendChild: function(row, cell) {
```
Comment (San4es) [inline js/ui/calendar/ui.calendar.base_view.js]:
```suggestion
    _appendCell: function(row, cell) {
```
?

File: js/ui/calendar/ui.calendar.base_view.js [hunk 4]
```diff
@@ -86,49 +86,39 @@ const BaseView = Widget.inherit({
...
-        const renderCell = (cellIndex) => {
-            // T425127
-            if(prevCellDate) {
-                fixTimezoneGap(prevCellDate, cellDate);
-            }
-
-            prevCellDate = cellDate;
+        for(let indexRow = 0, len = this.option('rowCount'); indexRow < len; indexRow++) {
+            rowData.row = this._createRow();
+            this._iterateCells(colCount, this._renderCell.bind(this, rowData));
```
Comment (San4es) [inline js/ui/calendar/ui.calendar.base_view.js]:
```suggestion
            for(const colIndex = 0, colCount = this.option('colCount'); colIndex < colCount; colIndex++) {
                this._renderCell(rowData, colIndex);
            }
```

File: js/ui/calendar/ui.calendar.base_view.js [hunk 5]
```diff
@@ -178,11 +192,11 @@ const BaseView = Widget.inherit({
         };
     },
 
-    _iterateCells: function(colCount, delegate) {
+    _iterateCells: function(colCount, callback) {
```
Comment (San4es) [inline js/ui/calendar/ui.calendar.base_view.js]: Get rid of it?

See https://github.com/DevExpress/DevExtreme/pull/11846#discussion_r374634953

File: js/ui/calendar/ui.calendar.base_view.js (hunk 2 above)
Comment (DokaRus) [reply js/ui/calendar/ui.calendar.base_view.js]: Saved the use of `let` - we iterate the value of this variable.

File: js/ui/calendar/ui.calendar.base_view.js (hunk 5 above)
Comment (DokaRus) [reply js/ui/calendar/ui.calendar.base_view.js]: Also removed the code duplication when rendering the header row of the month view

**Review by San4es — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**